File: v26_mediator/java/AirMain1.java
package v26_mediator.java;

class Airplane {
    private String name;

    public Airplane(String name) {
        this.name = name;
    }

    public void send(String message, Airplane receiver) {
        System.out.println(this.name + " sends message to " + receiver.getName() + ": " + message);
        receiver.receive(message, this);
    }

    public void receive(String message, Airplane sender) {
        System.out.println(this.name + " receives message from " + sender.getName() + ": " + message);
    }

    public String getName() {
        return this.name;
    }
}

public class AirMain1 {
    public static void main(String[] args) {
        Airplane airplane1 = new Airplane("Airplane1");
        Airplane airplane2 = new Airplane("Airplane2");
        Airplane airplane3 = new Airplane("Airplane3");

        airplane1.send("Hello, Airplane2!", airplane2);
        airplane2.send("Hello, Airplane3!", airplane3);
        airplane3.send("Hello, Airplane1!", airplane1);
    }
}

File: v26_mediator/python/air_main1.py
class Airplane:
    def __init__(self, name):
        self.name = name

    def send(self, message, receiver):
        print(f'{self.name} sends message to {receiver.name}: {message}')
        receiver.receive(message, self)

    def receive(self, message, sender):
        print(f'{self.name} receives message from {sender.name}: {message}')


if __name__ == '__main__':
    airplane1 = Airplane('Airplane1')
    airplane2 = Airplane('Airplane2')
    airplane3 = Airplane('Airplane3')

    airplane1.send('Hello, Airplane2!', airplane2)
    airplane2.send('Hello, Airplane3!', airplane3)
    airplane3.send('Hello, Airplane1!', airplane1)

